src/utils/api-request.ts

```diff
@@ -174,6 +207,57 @@ export class HttpClient {
           `Error while making request: ${err.message}. Error code: ${err.code}`);
       });
   }
+
+  private createHttpResponse(resp: LowLevelResponse): HttpResponse {
+    if (resp.multipart) {
+      return new MultipartHttpResponse(resp);
+    }
+    return new DefaultHttpResponse(resp);
+  }
+}
+
+/**
+ * Parses a full HTTP response message containing both a header and a body.
+ *
+ * @param {string|Buffer} response The HTTP response to be parsed.
+ * @param {HttpRequestConfig} config The request configuration that resulted in the HTTP response.
+ * @return {HttpResponse} An object containing the parsed HTTP status, headers and the body.
+ */
+export function parseHttpResponse(
+  response: string | Buffer, config: HttpRequestConfig): HttpResponse {
+
+  const responseText: string = validator.isBuffer(response)
+    ? response.toString('utf-8') : response as string;
+  const endOfHeaderPos: number = responseText.indexOf('\r\n\r\n');
+  const headerLines: string[] = responseText.substring(0, endOfHeaderPos).split('\r\n');
+
+  const statusLine: string = headerLines[0];
+  const status: string = statusLine.trim().split(/\s/)[1];
```

Comment:
Validate that the array has >=2 elements and before that headerLines has at least 1 element.

Reply:
`slice(1)` returns an empty array when length is less than 2. So it's safe to iterate over it without a length check. Similarly, `headerLines` is the result of a `split()` call, which always returns an array with at least 1 element.

Reply:
We use split and slice for `headerLines`. For `statusLine` I added a check to cover the case of a malformed header. Although chances of it happening should be negligibly small.